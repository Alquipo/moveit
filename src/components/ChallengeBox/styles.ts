import styled, { css } from 'styled-components'
import { darken } from 'polished'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    height: 100%;

    background-color: ${theme.colors.backgroundBox};
    border-radius: 5px;
    box-shadow: ${theme.boxShadow};

    transition: background-color 0.3s, box-shadow 0.3s;

    padding: 1.5rem 2rem;

    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;

    text-align: center;

    ${media.lessThan('medium')`
    margin-bottom: 4rem;

  `}
  `}
`
export const ChallengeActive = styled.div`
  ${({ theme }) => css`
    height: 100%;
    display: flex;
    flex-direction: column;

    header {
      color: ${theme.colors.blue};
      transition: color 0.3s;

      font-weight: 600;
      font-size: 1.25rem;
      padding: 0 2rem 1.5rem;
      border-bottom: 1px solid ${theme.colors.grayLine};
    }

    main {
      flex: 1;
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center;

      strong {
        font-size: 2rem;
        font-weight: 600;
        color: ${theme.colors.title};
        transition: color 0.3s;

        margin: 1.5rem 0 1rem;
      }

      p {
        line-height: 1.5;
      }
    }

    footer {
      display: grid;
      grid-template-columns: 1fr 1fr;
      gap: 1rem;
    }
  `}
`

type ChallengeButtonProps = {
  failed?: boolean
  succeeded?: boolean
}
export const ChallengeButton = styled.button<ChallengeButtonProps>`
  ${({ theme, failed, succeeded }) => css`
    height: 3rem;

    display: flex;
    align-items: center;
    justify-content: center;

    border: 0;
    border-radius: 5px;

    color: ${theme.colors.backgroundBox};
    transition: color 0.3s;

    font-size: 1rem;
    font-weight: 600;

    outline: 0;

    ${failed &&
    css`
      background-color: ${theme.colors.red};
      transition: background-color 0.3s;

      &:hover {
        background-color: ${darken(0.05, theme.colors.red)};
      }
    `}

    ${succeeded &&
    css`
      background-color: ${theme.colors.green};
      transition: background-color 0.3s;

      &:hover {
        background-color: ${darken(0.1, theme.colors.green)};
      }
    `}
  `}
`

export const ChallengeNotActive = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    color: ${theme.colors.text};

    strong {
      font-size: 1.5rem;
      font-weight: 500;
      line-height: 1.4;
    }

    p {
      display: flex;
      flex-direction: column;
      align-items: center;
      line-height: 1.4;
      margin-top: 3rem;

      img {
        margin-bottom: 1rem;
      }
    }
  `}
`
