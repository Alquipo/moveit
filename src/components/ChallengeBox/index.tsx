import * as S from './styles'

const ChallengeBox = () => {
  const hasActiveChallenge = true

  return (
    <S.Wrapper>
      {hasActiveChallenge ? (
        <S.ChallengeActive>
          <header>Ganhe 400 xp</header>

          <main>
            <img src="icons/body.svg" alt="challenge Image" />
            <strong>Novo desafio</strong>
            <p>Levante e fa√ßa uma caminhada de 3 minutos</p>
          </main>

          <footer>
            <S.ChallengeButton failed type="button">
              Falhei
            </S.ChallengeButton>
            <S.ChallengeButton succeeded type="button">
              Completei
            </S.ChallengeButton>
          </footer>
        </S.ChallengeActive>
      ) : (
        <S.ChallengeNotActive>
          <strong>Finalize um ciclo para receber um desafio</strong>
          <p>
            <img src="icons/level-up.svg" alt="Level Up" />
            Avance de level completando os desafios.
          </p>
        </S.ChallengeNotActive>
      )}
    </S.Wrapper>
  )
}

export default ChallengeBox
