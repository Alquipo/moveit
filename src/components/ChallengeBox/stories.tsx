import { Story, Meta } from '@storybook/react/types-6-0'
import ChallengeBox from '.'

export default {
  title: 'ChallengeBox',
  component: ChallengeBox
} as Meta

export const Default: Story = () => <ChallengeBox />
