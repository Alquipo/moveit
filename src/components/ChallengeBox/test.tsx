import { render, screen } from '@testing-library/react'

import ChallengeBox from '.'

describe('<ChallengeBox />', () => {
  it('should render the heading', () => {
    const { container } = render(<ChallengeBox />)

    expect(
      screen.getByRole('heading', { name: /ChallengeBox/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
