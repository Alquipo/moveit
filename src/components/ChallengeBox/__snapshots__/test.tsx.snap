// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChallengeBox /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    ChallengeBox
  </h1>
</main>
`;
